### PR TITLE
[saas-file-validator] add validation for unique saas file and env combination

### PR DESCRIPTION
**Pull request by maorfr**
part of https://issues.redhat.com/browse/APPSRE-3342

we are deploying using Tekton PipelineRuns, which need to have a name of 63 chars or less (kubernetes limitation).
since we add a timestamp, we need to generate a unique combination in 51 chars to avoid conflicts between deployments.

### CODE REVIEW

File: reconcile/utils/saasherder.py [hunk 1]
```diff
@@ -178,6 +184,24 @@ def _validate_saas_files(self):
             for duplicate in duplicates:
                 logging.error(msg.format(duplicate))
 
+    def _check_saas_file_env_combo_unique(self, saas_file_name, env_name):
+        # max tekton pipelinerun name length can be 63.
+        # leaving 12 for the timestamp leaves us with 51
+        # to create a unique pipelinerun name
+        tkn_long_name = f"{saas_file_name}-{env_name}"
+        tkn_name = tkn_long_name[:50]
```
Comment (Piojo) [inline reconcile/utils/saasherder.py:192]: Make this 50 a constant, please. Magic numbers end up biting us.

**Review by jfchevrette — APPROVED** [review]
LGTM

**Review by Piojo — APPROVED** [review]
Please, fix the nit on `50`, but otherwise...

**Comment by maorfr** [comment]
hold merging until https://gitlab.cee.redhat.com/service/app-interface/-/merge_requests/21127 is merged

File: reconcile/utils/saasherder.py (hunk 1 above)
Comment (maorfr) [reply reconcile/utils/saasherder.py]: done

**Pull request merged**